David/855 r nct 2

haveanicedavid opened the pull request:
## Description

Should be merged first: https://github.com/regen-network/regen-sanity/pull/14

Closes: regen-network/regen-registry#855

Live preview: https://deploy-preview-869--regen-website.netlify.app/nct/

EDIT 3/21:

This has been approved by Erika and Sarah so should be good to merge once it gets enough reviews, but it's not looking like it'll be done tonight and I'm out of town starting tomorrow morning. 

Whoever reviews this can merge, once the relevant Sanity PR has also been merged (and `yarn deploy` is run from the sanity repo so the prod data gets updated before)

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles 
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

haveanicedavid commented on tsconfig.json:
```diff
@@ -14,6 +14,7 @@
     "pretty": true,
     "removeComments": false,
     "sourceMap": true,
+    "strictNullChecks": true,
```
I added this do to the [recommended settings of material UI](https://github.com/regen-network/regen-sanity/pull/14) to try and figure out a bug - I don't think it was related, but leaving just because it's recommended

haveanicedavid commented on web-components/src/components/block-content/BlockContent.tsx:
```diff
@@ -0,0 +1,86 @@
+  tooltipText?: string;
+  noYMargin?: boolean;
+  /** we can't pass `sx` directly, so use wrapper to target elements */
+  sxWrap?: SxProps<Theme>;
```
I was hoping Git handled this rename better... I moved the logic out of the main `index` file just because I had several `index` files open in my editor and prefer named components. Can easily revert if needed. 

The only new part here is `sxWrap` - used to target the rendered content via `sx`

victordelval replied:
yes, it is also my preference to avoid index files so that they contain code... at most to export

haveanicedavid commented on web-components/src/components/blog-post/index.tsx:
```diff
@@ -77,6 +75,7 @@ const BlogPost: React.FC<BlogPostProps> = ({
         )}
       </Typography>
       <OutlinedButton
+        size="small"
```
this was a bit out of scope, but I was frustrated that our patterns weren't in line with some of MUI's suggestions - specifically with the button's `size` attribute. Before this PR, it didn't really affect anything but font size (and not on all breakpoints) - now `small`, `medium` and `large` render buttons in accordance with the figma styleguide, so we shouldn't have to pass so many custom styles to buttons. I'll point out a few relevant areas...

haveanicedavid commented on web-components/src/components/buttons/ContainedButton.tsx:
```diff
@@ -1,43 +1,28 @@
 
-const CustomButton = withStyles((theme: Theme) => ({
-  root: {
-    borderRadius: '2px',
-    fontFamily: theme.typography.h1.fontFamily,
-    fontWeight: 800,
-    color: theme.palette.primary.main,
-    letterSpacing: '1px',
-    boxShadow: 'none',
-    whiteSpace: 'nowrap',
-    [theme.breakpoints.down('sm')]: {
```
most of these are moved to the `muiTheme.ts`'s `component` section `styleOverrides`

haveanicedavid commented on web-components/src/components/sliders/ResponsiveSlider.tsx:
```diff
@@ -81,7 +81,7 @@ const useStyles = makeStyles<Theme, StyleProps>((theme: Theme) => ({
         display: 'flex',
         '& > div': {
-          width: '100%',
+          width: '97%',
```
render cards without cutting off right border

haveanicedavid commented on web-registry/src/components/organisms/CreditBatches.tsx:
```diff
@@ -98,9 +98,11 @@ const CreditBatches: React.FC<CreditBatchProps> = ({
+        .then(sortableBatches => {
+          setBatches(sortableBatches.data);
+        })
+        .catch(console.error); // eslint-disable-line no-console
```
because the REST methods are throwing errors on failed requests, if we don't handle those errors wherever the fetch methods are called, they can crash the app (like this did)

haveanicedavid commented on web-www/src/components/BackgroundSection.tsx:
```diff
@@ -196,33 +205,22 @@ const BackgroundSection = ({
+    <BackgroundImage
+      Tag="section"
+      className={clsx(className, classes.root)}
+      fluid={imageDataMobile && isMobile ? imageDataMobile : imageData}
```
majority of the deleted above was repeated code, aside from passing `imageDataMobile` - this just DRYs up the code a bit and removes the unnecessary `Box`es (most of the diff here is prettier formatting)

haveanicedavid commented on web-www/src/sections/nct/LedgerSection.tsx:
```diff
@@ -0,0 +1,94 @@
+
+const query = graphql`
+  query nctLedgerSection {
+    bg: file(relativePath: { eq: "topo-bg-top.png" }) {
+      childImageSharp {
+        fluid(quality: 90) {
+          ...GatsbyImageSharpFluid_withWebp
+        }
+      }
+    }
```
I thought about moving the topo background into sanity, but it seems like overkill. This could probably be simplified into a hook or css class (as we're doing in the registry app)

haveanicedavid commented on web-www/src/sections/nct/TokenSection.tsx:
```diff
@@ -0,0 +1,215 @@
+  const data = sanityNctPage?.tokenSection;
+  const launchDate = sanityNctPage?.launchDate;
+
+  const btnPad = { px: { xs: 8, sm: 12 } } as const;
```
note: if you want to extract `sx` styles into a variable, you need to declare it `as const` to satisfy typescript

mhagel replied:
nice tip

haveanicedavid commented on web-components/src/components/header/NavLink.tsx:
```diff
@@ -26,16 +26,18 @@ export const useNavLinkStyles = makeStyles<Theme, { isActive: boolean }>(
 
 export type NavLinkProps = {
   href: string;
+  pathname: string;
```
I'm not sure why this hasn't happened before, but something about having the `nct` nav link in the highest level (not handled through one of the dropdown helper components like the others) messed up gatsby's build - there's no `window` object at build time, so the `window.location.pathname` check errored.

This is a workaround, but i'm also noticing i'm not very happy with the helper functions I built for nav - initially they were a sort of hack to work the new design into an older one, but the entire nav setup seems ripe for refactoring

mhagel reviewed (approved):
Looks really great. Just one `console.log` that probably ought to go away.

blushi commented on web-www/src/components/MarketingNav.tsx:
```diff
@@ -92,20 +93,44 @@ const MarketingNav: React.FC<{ location: Location }> = ({ location }) => {
+    {
+      title: 'NCT',
+      href: '/nct/',
+      isNew: true,
```
maybe this could be handled through sanity instead?

haveanicedavid replied:
removed for now in lieu of [Add rNCT button and header graphic to registry home page](https://app.zenhub.com/workspaces/regen-registry-5f8998bec8958d000f4609e2/issues/regen-network/regen-registry/856)

blushi reviewed (commented):
Really nice work, just some small comments

blushi commented:
@haveanicedavid shouldn’t all the “learn more” and “tech docs” buttons links open up in a new tab?
It seems like they are set up this way on sanity but somehow they do not open in a new tab on the deploy preview.

blushi reviewed (changes requested):
@haveanicedavid shouldn’t all the “learn more” and “tech docs” buttons links open up in a new tab?
It seems like they are set up this way on sanity but somehow they do not open in a new tab on the deploy preview.

haveanicedavid commented on web-components/src/components/header/HeaderDropdownItems.tsx:
```diff
@@ -32,6 +32,7 @@ const useStyles = makeStyles(theme => {
 export type HeaderDropdownItemProps = {
   title: string;
   href: string;
+  pathname: string;
```
I had to do some prop drilling with this... there is probably a cleaner solution, but this works for now

haveanicedavid commented on web-components/src/components/header/index.tsx:
```diff
@@ -107,9 +107,8 @@ const useStyles = makeStyles<Theme, StyleProps>((theme: Theme) => {
-        '& span:hover, svg:hover, path:hover': {
-          borderBottom: 'none',
-        },
+        display: 'flex',
+        alignSelf: 'baseline',
```
without this, the icons weren't lining up vertically.

haveanicedavid commented on web-www/src/sections/nct/LedgerSection.tsx:
```diff
@@ -0,0 +1,95 @@
+            <OutlinedButton
+              sx={{ px: [8, 12], mt: [4, 8] }}
+              href={data?.button?.buttonLink?.buttonHref || ''}
+              target={data?.button?.buttonBlankTarget ? '_blank' : '_self'}
```
I wasn't passing `target` before - this should fix (also added elsewhere)

blushi replied:
It'd be great to have that coded at the button component level so we don't have to repeat this logic every time, could you create a code-hygiene issue to track that?
we have some kind of util function already `onBtnClick` but it's on web-registry for now so should be adapted to work for both CRA and Gastby.

haveanicedavid replied:
@blushi sorry I missed this comment before merging! 

Looking at the current helpers, they aren't using react router so the site will reload upon clicking. Created a story and added some thoughts [here](https://app.zenhub.com/workspaces/regen-registry-5f8998bec8958d000f4609e2/issues/regen-network/regen-registry/881)

haveanicedavid commented:
@blushi I believe I addressed all of your comments, but did have to pass `pathname` down to `NavLink` which isn't super clean, but should work for now

haveanicedavid commented:
@Mergifyio backport redwood hambach
